From 41bbf5401c9d4a1f594fd823d9383e52b98d0a2b Mon Sep 17 00:00:00 2001
From: Devyn Liu <liudingyuan@huawei.com>
Date: Thu, 11 Apr 2024 21:27:44 +0800
Subject: [PATCH 606/960] downstream: spi: Add verification for the
 max_frequency provided by the firmware

Add verification for the max_frequency provided by the firmware.
If the value is 0, the parameter is regarded as invalid and an
error code is returned.

Signed-off-by: Devyn Liu <liudingyuan@huawei.com>
Signed-off-by: Junhua Lu <lujunhua7@h-partners.com>
Signed-off-by: huwentao <huwentao19@h-partners.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/spi/spi-hisi-kunpeng.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-hisi-kunpeng.c b/drivers/spi/spi-hisi-kunpeng.c
index 16054695bdb0..fb8b825d52de 100644
--- a/drivers/spi/spi-hisi-kunpeng.c
+++ b/drivers/spi/spi-hisi-kunpeng.c
@@ -481,8 +481,11 @@ static int hisi_spi_probe(struct platform_device *pdev)
 		return -EINVAL;
 	}
 
-	if (host->max_speed_hz == 0)
-		return dev_err_probe(dev, -EINVAL, "spi-max-frequency can't be 0\n");
+	if (host->max_speed_hz == 0) {
+		dev_err(dev, "invalid max SPI clocking speed, max-freq=%u\n",
+				host->max_speed_hz);
+		return -EINVAL;
+	}
 
 	ret = device_property_read_u16(dev, "num-cs",
 					&host->num_chipselect);
-- 
2.43.0

